SpatialGDK/Source/SpatialGDKFunctionalTests/Private/TestMapGeneration.cpp

```diff
@@ -52,22 +54,36 @@ bool GenerateTestMaps()
 	TArray<UClass*> TestMapClasses;
 	for (TObjectIterator<UClass> Iter; Iter; ++Iter)
 	{
-		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) && *Iter != UGeneratedTestMap::StaticClass())
+		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) || Iter->IsChildOf(ASpatialFunctionalTest::StaticClass()))
 		{
 			TestMapClasses.Add(*Iter);
 		}
 	}
 
 	for (UClass* TestMapClass : TestMapClasses)
 	{
-		UGeneratedTestMap* TestMap = NewObject<UGeneratedTestMap>(GetTransientPackage(), TestMapClass);
-		TestMap->AddToRoot(); // Okay, must admit, not completely sure what's going on here, seems like even though the commandlet is
-							  // the outer of the newly generated object, the object still gets GCed when creating a new map, so have to add
-							  // to root here to prevent GC
+		IGeneratableTestMap* TestMap;
+		if (TestMapClass->IsChildOf(ASpatialFunctionalTest::StaticClass()))
+		{
+			TestMap = NewObject<ASpatialFunctionalTest>(GetTransientPackage(), TestMapClass);
+		}
+		else
+		{
+			TestMap = NewObject<UGeneratedTestMap>(GetTransientPackage(), TestMapClass);
+		}
```

Comment:
Unreal isn't happy with `IGeneratableTestMap* TestMap = NewObject<IGeneratableTestMap>(GetTransientPackage(), TestMapClass);` because interface, is there way to make something clean - I couldn't find API that just took UCLASS 😞